src/api/spec/controllers/webui/projects/pulse_controller_spec.rb

```diff
@@ -0,0 +1,57 @@
+RSpec.describe Webui::Projects::PulseController do
+  let(:project) { create(:project, name: 'pulse_project') }
+  let(:package) { create(:package_with_file, name: 'pulse_package', project: project) }
+  let(:first_user) { create(:confirmed_user) }
+  let(:second_user) { create(:confirmed_user) }
+
+  describe '#show' do
+    render_views
+
+    before do
+      get :show, format: :html, params: { project_name: project.name }
+    end
+
+    it 'load the whole page' do
+      expect(response.body).to have_css('#range-header')
+      expect(response.body).to have_css('#pulse')
+    end
+
+    it 'assigns the correct instance variables' do
+      expect(controller.instance_variable_get(:@range)).to eq('week')
+      expect(controller.instance_variable_get(:@builds)).to be_nil
+    end
+  end
+
+  describe '#show async' do
+    before do
+      10.times do
+        Event::BuildSuccess.create(
+          project: project.name,
+          package: package.name,
+          reason: 'build'
+        )
+      end
+      10.times do
+        Event::BuildFail.create(
+          project: project.name,
+          package: package.name,
+          reason: 'build'
+        )
+      end
+      10.times do
+        Event::CommentForProject.create(
+          project: project.name,
+          commenter: second_user.id,
+          comment_body: "Hey #{first_user.login}, how are you?"
+        )
+      end
+
+      post :show, format: :js, params: { project_name: project.name, range: 'month' }
+    end
+
+    it 'assigns the correct instance variables' do
+      expect(controller.instance_variable_get(:@builds).count).to eq(20)
```

Comment:
What are you testing here?

Reply:
Testing the fact that `@builds` is properly evaluated, which means the code path went through the `set_pulse` method, which is supposed to happen only in the `async` call.
It could just be `.not_to be_nil` (the opposite of [line 21](https://github.com/openSUSE/open-build-service/pull/17730/files#diff-bc1b80a6bd367ccad8f5f2a530b493f563a57e0fc4a7664dc18a863e090dd101R21)), but why not to make sure it is also set correctly.

Reply:
https://github.com/openSUSE/open-build-service/tree/master/src/api/spec#what-to-test-and-how

Reply:
Alright. Dropped redundant test, doing the first scenario is enough